src/utils/jsonlogic.ts

```diff
@@ -0,0 +1,169 @@
+/**
+ * JsonLogic type checking utility functions
+ */
+import {infer} from '@open-formulieren/infernologic';
+import {type JSONObject, type JSONValue} from '@open-formulieren/types/lib/types';
+
+import {type AnyComponentSchema} from '@/types';
+
+/**
+ * @param {object | array | number | string} logic - JsonLogic expression
+ * @param {object | array | number | string} expected - example value from expected result type e.g. [["label", "value"]]
```

Comment:
I've never understood how the JSDoc type definitions stack up against the actual type definitions in the code itself. To me it seems a risk of going out of sync.

The types here don't account for `null` which is a valid JSON type and I can't predict how things behave if I pass `null` (since the TS compiler will likely allow it).

Additionally, the `object` will probably allow me to pass `function` references :)